rails/lib/publishable.rb

```diff
@@ -1,37 +1,24 @@
-require 'aasm'
-#
-#  Methods for containers that can be made public through some workflow
-#  (challenge will be defining this in a project agnostic way...)
-#
+
 module Publishable
   ##
   ## Called when a class extends this module:
   ##
   def self.included(clazz)
     ## add before_save hooks
     clazz.class_eval do
-      # use rubyist-aasm gem (acts_as_state_machine)
-      # for publication status:
-      # see: http://www.practicalecommerce.com/blogs/post/440-Acts-As-State-Machine-Is-Now-A-Gem
-      # and http://www.ruby-forum.com/topic/179721
-      # for a discussion on how the new aasm gem differs from the old plugin...
-      include AASM
-
-      aasm_initial_state :draft
-      aasm_column :publication_status
+
       @@protected_publication_states=[:published]
       @@publication_states = [:draft,:published,:private]
-      @@publication_states.each { |s| aasm_state s}
 
       # this needs to come after the class variable definition...
       cattr_accessor :publication_states
 
-      aasm_event :publish do
-        transitions :to => :published, :from => [:draft]
+      def publish!
```

Comment:
I just looked at the test results and it looks like you also need to add a `publish` method:

```
undefined method `publish' for #<ExternalActivity:0x0000000006223a48>
Did you mean?  publish!
               published?
               public?
/home/travis/.rvm/gems/ruby-2.3.7/gems/activemodel-4.0.13/lib/active_model/attribute_methods.rb:439:in `method_missing'
/home/travis/.rvm/gems/ruby-2.3.7/gems/activerecord-4.0.13/lib/active_record/attribute_methods.rb:168:in `method_missing'
/home/travis/build/[secure]-consortium/rigse/rails/lib/mock_data/create_default_data.rb:661:in `block in create_default_study_materials'
```